src/context/auth.js

```diff
@@ -33,13 +42,15 @@ const AuthProvider = ({ children }) => {
     }
 
     localStorage.setItem('token', res.data.token);
+    setToken(res.data.token);
 
-    setToken(res.token);
-    navigate(location.state?.from?.pathname || '/');
```

Comment:
this logic here was there so that, if we had visited /profile/12 page being LOGGED out, then on login, we would be redirected back to page /profile/12.

The BUG is that if no location exists, we just get redirected onto `/`.